### PR TITLE
🐳 Next generation of Docker CI

### PR DESCRIPTION
**Pull request by favonia**

### CODE REVIEW

**Comment by favonia** [comment]
Tags:
- base
- builder-<branch name>
- edge

**Review by jonsterling — COMMENTED** [review]
I don't think I'll be able to review this PR effectively because I'm not enough of an expert in build methodology, etc. But could you summarize what changes (from a developer's perspective, but not a build engineer's perspective) between `main` and this branch? Thanks!
